### DISCUSSION TITLE
Gunicorn multiple workers

### DISCUSSION THREAD

**Discussion by old-syniex** [Miksus/rocketry#97] [Q&A]
Hey just found out about this package and it looks awesome. 

I am running a FastAPI service with gunicorn.

I was wondering is there a way to make only one of the workers schedule a task? 

Thanks

**Comment by Miksus** [comment]
Hi, and thanks! Always nice to hear others see it interesting as well!

I have made a FastAPI implementation using Uvicorn. Perhaps this helps: https://github.com/Miksus/rocketry-with-fastapi (clone and use it as you wish)

I haven't yet tested how it could be integrated with Gunicorn though but I suppose something similar as I did with Uvicorn could work. I at least realized that Uvicorn has its own exiting behaviour in which I had to hook Rocketry. Otherwise, Rocketry has no information that keyboard interruption was called.

Do you manage with Uvicorn or did my answer provide any help? Unfortunately, I think pushing the newest update onward takes too much time at the moment to test other things out.

**Reply by gegnew** [reply]
@Miksus do you have any more ideas on running Rocketry with Gunicorn? I'm running Rocketry on the FastAPI event loop, but I haven't been able to come up with a way to prevent it running tasks on all Gunicorn workers.

**Reply by Miksus** [reply]
@gegnew I have been working on Rocketry integration with various web frameworks here: https://github.com/Miksus/subsystems

It seems I said "Gunicorn is experimental" in the docs and cannot remember exactly where I was left. I just remember reading a lot of Gunicorn code and realized it's quite a mess and hard to customize/extend under the hood. I'm planning to visit that project at some point again so I could take another look at how it was.

One problem was also that I could not really prototype it on my machine which is Windows (I have dual boot but then I cannot use my monitor thanks to Nvidia...).
